OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs

```diff
@@ -55,49 +82,78 @@ public OptimizelyConfigService(ProjectConfig projectConfig)
 
             foreach (Experiment experiment in experiments)
             {
-                var variationsMap = new Dictionary<string, OptimizelyVariation>();
-                foreach (Variation variation in experiment.Variations)
-                {
-                    var variablesMap = MergeFeatureVariables(projectConfig,
-                        featureVariableIdMap,
-                        experiment.Id,
-                        variation);
+                var variationsMap = GetVariationsMap(experiment, featureVariableIdMap, projectConfig);
 
-                    var optimizelyVariation = new OptimizelyVariation(variation.Id,
-                        variation.Key,
-                        variation.FeatureEnabled,
-                        variablesMap);
-
-                    variationsMap.Add(variation.Key, optimizelyVariation);
-                }
                 var optimizelyExperiment = new OptimizelyExperiment(experiment.Id,
                     experiment.Key,
+                    GetExperimentAudiences(experiment, projectConfig),
```

Comment:
place it in a separate variable and pass that variable as an arg. otherwise, it will cause problem in case of debugging.